src/idl_gen_general.cpp

```diff
@@ -587,6 +587,19 @@ std::string GenGetter(const Type &type) {
   }
 }
 
+// Returns the function name that is able to read a value of the given type.
+std::string GenGetterForLookupByKey(const Type &type,
+                                    const std::string data_buffer) {
+  auto getter = data_buffer + "." + FunctionStart('G') + "et";
+  if (type.base_type == BASE_TYPE_BOOL) {
+    getter = "0!=" + getter;
+  }
+  else if (GenTypeBasic(type, false) != "byte") {
```

Comment:
`GenTypeBasic(type, false) != "byte"` -> `SizeOf(type.base_type) != 1` ?
What does this code do? I don't understand how the `+=` works. I know we can't have test cases for keys of every possible type, but can you verify this generates legit code?